### PR TITLE
BUG: Avoid intercepting TypeError in DataFrame.agg

### LINKED ISSUE #49399: Bug: Don't intercept TypeError in DataFrame.agg
https://github.com/pandas-dev/pandas/blob/3370c8179de16b3f0556891a1ff06131e186dfd1/pandas/core/apply.py#L735-L745

We intercept any TypeError and reword the message to be about the DataFrame constructor. This likely made sense before, but with the code being refactored and #43741, TypeErrors can occur for many different reasons. We should just remove the `except` block above.

### PR DESCRIPTION
**Pull request by yuvanist**
- [x] closes #49399
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/apply/test_frame_apply.py [hunk 1]
```diff
@@ -1182,7 +1182,7 @@ def test_agg_multiple_mixed_raises():
 
     # sorted index
     # TODO: GH#49399 will fix error message
```
Comment (jbrockmendel) [inline pandas/tests/apply/test_frame_apply.py:1184]: is the TODO in this comment addressed? if so, can you remove it

File: pandas/tests/apply/test_frame_apply.py [hunk 2]
```diff
@@ -1292,7 +1292,7 @@ def test_nuiscance_columns():
     tm.assert_series_equal(result, expected)
 
     # TODO: GH#49399 will fix error message
```
Comment (jbrockmendel) [inline pandas/tests/apply/test_frame_apply.py:1294]: same here

File: pandas/tests/apply/test_frame_apply.py (hunk 1 above)
Comment (yuvanist) [reply pandas/tests/apply/test_frame_apply.py]: Yes, this is addressed and removed it.

File: pandas/tests/apply/test_frame_apply.py (hunk 2 above)
Comment (yuvanist) [reply pandas/tests/apply/test_frame_apply.py]: This is also removed.

**Review by rhshadrach — APPROVED** [review]
lgtm

**Review by jbrockmendel — APPROVED** [review]

**Comment by jbrockmendel** [comment]
thanks @yuvanist

**Pull request merged**